File: python/buttonLED.py
from time import sleep
import RPi.GPIO as GPIO
GPIO.setmode(GPIO.BCM)

LED = 17
button = 25
GPIO.setup(LED, GPIO.OUT)
GPIO.setup(button, GPIO.IN)

toggle = False
while True:
	GPIO.output(LED, toggle)
	
	if GPIO.input(button):
		toggle = not toggle
		print("Button pressed")
		while GPIO.input(button):	
			pass

File: wiringPi/buttonLED/buttonLED.c
#include <wiringPi.h>
#include <stdio.h>

const int LED = 0;
const int button = 6;

void setup()
{
	pinMode(LED, OUTPUT);
	digitalWrite(LED, 0);
    pinMode(button, INPUT);
}

int main(void)
{
	if(wiringPiSetup() == -1)
		return 1;
	
    bool toggle = false;

	while(1)
	{
        digitalWrite(LED, toggle);

		if (digitalRead(button))
            toggle = !toggle;
            while(digitalRead(button))

	}
	return 0;
}
